colony/commands/sb.py

```diff
@@ -111,20 +120,39 @@ def do_start(self):
         inputs = parse_comma_separated_string(self.args["--inputs"])
         artifacts = parse_comma_separated_string(self.args["--artifacts"])
 
+        temp_working_branch = ""
+        repo = BlueprintRepo(os.getcwd())
+
         if branch:
             working_branch = branch
         else:
             try:
-                working_branch = get_blueprint_working_branch(os.getcwd(), blueprint_name=bp_name)
+                working_branch = get_blueprint_working_branch(repo, blueprint_name=bp_name)
                 self.message(f"Automatically detected current working branch: {working_branch}")
+
             except BadBlueprintRepo as e:
                 working_branch = None
                 logger.warning(
                     f"No branch has been specified and it could not be identified from the working directory; "
                     f"reason: {e}. A branch of the Blueprints Repository attached to Colony Space will be used"
                 )
 
-        # TODO(ddovbii): This obtaining default values magic mast be refactored
+            # Checking if:
+            # 1) User has specified not use local (not remote)
+            # 2) User is in ana actual git dir (working_branch)
+            # 3) There is even a need to create a temp branch for out-of-sync reasons (repo.is_current_branch_synced())
+            if not remote and working_branch and not repo.is_current_branch_synced():
+                temp_working_branch = working_branch
+                try:
+                    temp_working_branch = switch_to_temp_branch(repo, working_branch)
+                except Exception as e:
+                    logger.error("Was not able to create temp branch.")
+                    if working_branch:
+                        logger.warning(f"Branch {working_branch} will be used. Reason: {str(e)}")
+                    else:
+                        logger.warning(f"Remote branch will be used. Reason: {str(e)}")
```

Comment:
this looks to be a duplicate with what we have in bp.py.
It will good to call the same code from both commands